synapse/federation/transport/server/_base.py

```diff
@@ -233,31 +239,37 @@ def __init__(
         self.ratelimiter = ratelimiter
         self.server_name = server_name
 
-    def _wrap(self, func):
+    def _wrap(
+        self, func: Callable[..., Awaitable[Tuple[int, JsonDict]]]
```

Comment:
As much as I dislike `Any`, I think the return type of `func` is `Awaitable[Tuple[int, Any]]` since there are a bunch of handlers in `synapse/federation/transport/server/federation.py` that return something other than a `JsonDict`.

Reply:
It really needs to be "something that can be serialized to JSON" I think... which we don't have a good type for. 😢 I can make it `Any` for now and we can improve if we'd like.

(I think the only things returned are `JsonDict` or `Tuple[int, JsonDict]` but there could be other weirder things.)

Reply:
Sounds good, I'd avoid trying to define Jsonable/JsonObject for now, to avoid scope creep.

Reply:
Oh, this seems to better match the return type of `_wrap` anyway. 👍